fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/ImmutableAvailability.java

```diff
@@ -97,11 +101,15 @@ public ImmutableAvailability(
                 entry -> new SimplifiedIntervalList(entry.getValue());
 
         Map<Column, List<Interval>> map = dimensionIntervals.entrySet().stream()
+                .filter(entry -> dimensionDictionary.findByApiName(entry.getKey()) != null)
                 .collect(Collectors.toMap(dimensionKeyMapper, valueMapper));
         map.putAll(
                 metricIntervals.entrySet().stream()
                         .collect(Collectors.toMap(metricKeyMapper, valueMapper))
         );
+        if (map.size() == 0) {
+            LOG.warn("No legal availability columns discovered when building availability instance.");
```

Comment:
We should include some info about this availability so that we have a sense of _what_ failed. The `tableName` is probably a good candidate.